### PR TITLE
move`useDebounced` and add `useMediaQuery`

### PR DESCRIPTION
**Pull request by 21christiansc**

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -2,3 +2,4 @@
 ============
 
 * The import path for `mount()` has changed `/mount.js -> /react/mount.mjs`.
+* The import path for `useDebounced` has cahnged `/debounce -> /react/hooks/debounce`.
```
Comment (keichinger) [inline UPGRADE.md]:
```suggestion
* The import path for `useDebounced` has changed `/debounce -> /react/hooks/debounce`.
```

File: src/react/hooks/media-query.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+	useEffect(() =>
+	{
+		const mediaQuery = window.matchMedia(query);
+		setActive(mediaQuery.matches);
+
+		const updater = (event: MediaQueryListEvent) => setActive(event.matches);
+		mediaQuery.addEventListener("change", updater);
+
+		return () => mediaQuery.removeEventListener("change", updater);
```
Comment (keichinger) [inline src/react/hooks/media-query.ts]: Can't we simplify that with the usage of `onOff()`?

```suggestion
		const mediaQuery = window.matchMedia(query);
		setActive(mediaQuery.matches);
		
		return onOff(mediaQuery, "change", (event: MediaQueryListEvent) => setActive(event.matches));
```

Comment (apfelbox) [reply src/react/hooks/media-query.ts]: We will lose support for Safari 13 with this implementation right now (as we have no fallback for the old implementation), which hit us in one of the projects earlier.

JFYI

**Review by apfelbox — APPROVED** [review]
See comment about no Safari 13- support. Is probably fine though

File: src/react/hooks/media-query.ts (hunk 2 above)
Comment (keichinger) [reply src/react/hooks/media-query.ts]: Wouldn't it be better to add support to `onOff()` instead of fixing it just in this one place?

Comment (apfelbox) [reply src/react/hooks/media-query.ts]: Adding support for `onOff` for `addListener`? That would be weird, as `onOff` is for events, and `addListener` is MediaQuery-specific.

**Pull request merged**